index.js

```diff
@@ -15,27 +13,53 @@ function getOrThrow(configField) {
 
     return value;
 }
-const githubToken = Buffer.from(getOrThrow('GITHUB_TOKEN')).toString('base64');
-const githubIdColumnNumber = getOrThrow('CSV_COLUMN_NUMBER_FOR_GITHUB_ID');
-const alternateIdColumnNumber = getOrThrow('CSV_COLUMN_NUMBER_FOR_ALTERNATE_ID');
-let githubImportantEvents = getOrThrow('GITHUB_IMPORTANT_EVENTS').split(',');
+const githubToken = Buffer.from(getOrThrowIfMissingOrEmpty('GITHUB_TOKEN')).toString('base64');
+const githubIdColumnNumber = getOrThrowIfMissingOrEmpty('CSV_COLUMN_NUMBER_FOR_GITHUB_ID');
+const alternateIdColumnNumber = getOrThrowIfMissingOrEmpty('CSV_COLUMN_NUMBER_FOR_ALTERNATE_ID');
+let githubImportantEvents = getOrThrowIfMissingOrEmpty('GITHUB_IMPORTANT_EVENTS').split(',');
+const timeZone = process.env.TIMEZONE;
 
 const ignoreSelfOwnedEvents = (process.env.IGNORE_SELFOWNED_EVENTS || 'false').toLowerCase();
 console.log(`Configuration set to ignore self-owned events? ${ignoreSelfOwnedEvents}`);
 if (ignoreSelfOwnedEvents !== 'true' && ignoreSelfOwnedEvents !== 'false') {
-    console.error(`IGNORE_SELFOWNED_EVENTS must be "true" or "false"`);
+    console.error('IGNORE_SELFOWNED_EVENTS must be "true" or "false"');
     process.exit(1);
 }
 
 //Helper Functions
+function createTimeZone(timeZoneIdentifier) {
+    if (!timeZoneIdentifier || timeZoneIdentifier === '') {
+        return IANAZone.create('UTC');
+    }
+    if (timeZoneIdentifier === 'local') {
+        return LocalZone.instance;
+    }
+    if (IANAZone.isValidZone(timeZoneIdentifier)) {
+        return IANAZone.create(timeZoneIdentifier);
+    }
+    console.error(
+        `Unknown time zone "${timeZoneIdentifier}". Fix the TIMEZONE entry of the .env file.`
+    );
+    process.exit(1);
+}
+
+function createLuxonMomentFromIso(isoDateTimeString, timeZoneIdentifier) {
+    const zone = createTimeZone(timeZoneIdentifier);
+
+    const moment = DateTime.fromISO(isoDateTimeString, {
+        zone,
+    });
+
+    return moment;
+}
+
 function parseDatesFromArgv() {
-    const timeZone = getOrThrow('TIMEZONE');
+    console.log(`Using time zone: ${createTimeZone(timeZone).name}`);
     const startDate = process.argv[2];
     const endDate = process.argv[3];
 
-    const startMoment = moment.tz(startDate, timeZone).startOf('day');
-
-    const endMoment = moment.tz(endDate, timeZone).endOf('day');
+    const startMoment = createLuxonMomentFromIso(startDate, timeZone).startOf('day');
```

Comment:
There's still something off about this. When I run the old code (Ingrid's recently merged PR) with 
`cat CSVsToParse/shortCSV.csv | node index.js 2020-12-08 2020-12-08` I get back (minus some PII)
```
Configuration set to ignore self-owned events? false
Users that contributed between Tue Dec 08 2020 00:00:00 GMT-0600 and Tue Dec 08 2020 23:59:59 GMT-0600 
japple...
alison...
dua...
dgel...
 ```

When I run that same command with your code, I just get
``` 
Configuration set to ignore self-owned events? false
Using time zone: Etc/GMT+6
Users that contributed between December 8, 2020, 12:00 AM GMT-6 and December 8, 2020, 11:59 PM GMT-6
 ```
When I change the dates to `2020-12-08 2020-12-09` and run your code, I get
```
Configuration set to ignore self-owned events? false
Using time zone: Etc/GMT+6
Users that contributed between December 8, 2020, 12:00 AM GMT-6 and December 9, 2020, 11:59 PM GMT-6 
japple...
dgel...
```
And `2020-12-07 2020-12-08` gives us
```
Configuration set to ignore self-owned events? false
Using time zone: Etc/GMT+6
Users that contributed between December 7, 2020, 12:00 AM GMT-6 and December 8, 2020, 11:59 PM GMT-6 
alison...
dua...
dgel...
```

Reply:
I believe my latest commits fix all of this. It was definitely not doing correct calculations before. Now, I believe it is, both for UTC and for non-UTC time zones. I added tests.